source/UF/H-Levels.lagda

```diff
@@ -163,108 +193,34 @@ The subuniverse of types of hlevel n is defined as follows.
+equiv-preserves-hlevel (succ n) {X} {Y} X-h-lev Y-h-lev =
+ hlevel-closed-under-embedding (succ n) ⋆ e'
+                               (hlevel-closed-under-Π (succ n) (λ _ → Y)
+                                                      (λ _ → Y-h-lev))
+ where
+  e' : (X ≃ Y) ↪ (X → Y)
+  e' = (⌜_⌝ , pr₁-is-embedding (λ f → being-equiv-is-prop fe' f))
 
 ℍ-is-of-next-hlevel : (n : ℕ)
                     → (𝓤 : Universe)
                     → is-univalent 𝓤
                     → (ℍ n 𝓤) is-of-hlevel (succ n)
-ℍ-is-of-next-hlevel zero 𝓤 ua = C
+ℍ-is-of-next-hlevel n 𝓤 ua (X , l) (Y , l') =
+ hlevel-closed-under-equiv n II (equiv-preserves-hlevel n l l')
  where
-  C : (X X' : ℍ zero 𝓤) → is-contr (X ＝ X')
-  C (X , X-is-contr) (X' , X'-is-contr) =
-    hlevel-closed-under-equiv zero ((X , X-is-contr) ＝ (X' , X'-is-contr))
-                              (X ≃ X') e C'
-   where
-    e = ((X , X-is-contr) ＝ (X' , X'-is-contr)) ≃⟨ ≃-sym (to-subtype-＝-≃
-                                                  (λ X → being-singleton-is-prop
-                                                         (fe 𝓤 𝓤))) ⟩
-        (X ＝ X')                                ≃⟨ univalence-≃ ua X X' ⟩
-        (X ≃ X')                                 ■
-    P : is-prop (X ≃ X')
-    P = ≃-is-prop fe (is-prop'-implies-is-prop
-                        (hlevels-are-upper-closed zero X' X'-is-contr))
-    C' : is-contr (X ≃ X')
-    C' = pointed-props-are-singletons (singleton-≃ X-is-contr X'-is-contr) P
-ℍ-is-of-next-hlevel (succ n) 𝓤 ua (X , l) (X' , l') =
-  hlevel-closed-under-equiv (succ n) ((X , l) ＝ (X' , l')) (X ≃ X') e
-      (hlevel-closed-under-embedding (succ n) ⋆ (X ≃ X') (X → X') e'
-                                     (hlevel-closed-under-Π (succ n) X
-                                                            (λ _ → X')
-                                                            (λ x x' → l' x')))
-  where
-   e = ((X , l) ＝ (X' , l')) ≃⟨ ≃-sym (to-subtype-＝-≃
-                                  (λ _ → Π-is-prop (fe 𝓤 𝓤)
-                                  (λ x → Π-is-prop (fe 𝓤 𝓤)
-                                  (λ x' → hlevel-relation-is-prop
-                                            n (x ＝ x'))))) ⟩
-       (X ＝ X')              ≃⟨ univalence-≃ ua X X' ⟩
-       (X ≃ X')               ■
-
-   e' : (X ≃ X') ↪ (X → X')
-   e' = (pr₁ , pr₁-is-embedding (λ f → being-equiv-is-prop fe f))
-
-\end{code}
-
-We now define the notion of a k-truncation using record types.
-
-\begin{code}
-
-record H-level-truncations-exist : 𝓤ω where
- field
-  ∣∣_∣∣_ : {𝓤 : Universe} → 𝓤 ̇ → ℕ → 𝓤 ̇
-  ∣∣∣∣-is-prop : {𝓤 : Universe} {X : 𝓤 ̇ } {n : ℕ} → is-prop (∣∣ X ∣∣ n)
-  ∣_∣_ :  {𝓤 : Universe} {X : 𝓤 ̇ } → X → (n : ℕ) → ∣∣ X ∣∣ n
-  ∣∣∣∣-rec : {𝓤 𝓥 : Universe} {X : 𝓤 ̇ } {Y : 𝓥 ̇ } {n : ℕ}
-           → Y is-of-hlevel n → (X → Y) → ∣∣ X ∣∣ n → Y
- infix 0 ∣∣_∣∣_
- infix 0 ∣_∣_
-
+  I = ≃-sym (to-subtype-＝-≃ (λ - → hlevel-relation-is-prop n -))
```

Comment:
Please swap the labels `I` and `II` and then put `II` in a `where`-clause of `I` so that its type is clear.